src/ImageGallery.jsx

```diff
@@ -580,27 +597,30 @@ export default class ImageGallery extends React.Component {
   _handleSwiping = (e, deltaX, deltaY, delta) => {
     const { galleryWidth, isTransitioning, scrollingUpDown } = this.state;
     const { swipingTransitionDuration } = this.props;
-    this._setScrollDirection(deltaX, deltaY);
-    if (!isTransitioning && !scrollingUpDown) {
-      const side = deltaX < 0 ? 1 : -1;
 
-      let offsetPercentage = (delta / galleryWidth * 100);
-      if (Math.abs(offsetPercentage) >= 100) {
-        offsetPercentage = 100;
-      }
+    this._withAnimationFrame(() => {
+      this._setScrollDirection(deltaX, deltaY);
+      if (!isTransitioning && !scrollingUpDown) {
+        const side = deltaX < 0 ? 1 : -1;
 
-      const swipingTransition = {
-        transition: `all ${swipingTransitionDuration}ms ease`
-      };
+        let offsetPercentage = (delta / galleryWidth * 100);
+        if (Math.abs(offsetPercentage) >= 100) {
+          offsetPercentage = 100;
+        }
 
-      this.setState({
-        offsetPercentage: side * offsetPercentage,
-        style: swipingTransition,
-      });
-    } else {
-      // don't move the slide
-      this.setState({ offsetPercentage: 0 });
-    }
```

Comment:
The only change here is that I wrapped this code with this._withAnimationFrame()